flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/source/PulsarSourceITCase.java

```diff
@@ -40,6 +42,9 @@
     @TestExternalSystem
     PulsarTestEnvironment pulsar = new PulsarTestEnvironment(PulsarRuntime.mock());
 
+    @Semantic
+    CheckpointingMode[] semantics = new CheckpointingMode[] {CheckpointingMode.EXACTLY_ONCE};
```

Comment:
I didn't find any ITCase for `CheckpointingMode.AT_LEAST_ONCE`,  we need add some case as we introduced the semantics  supports.

Reply:
Actually, this semantic tests are contained in the sink tests for Kafka connector. There are not ITCase in the source now.